### PR TITLE
Start post-Postgres migrations

### PR DESCRIPTION
**Pull request by c-du**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by dashrews78 — APPROVED** [review]
LGTM, but I'm not sure it will make it clearer for anyone else.  Though I do like the idea of the "Last" Rocks migration.

**Review by keyallis — COMMENTED** [review]
It might be good to add a little README or something just to give a more direct guidance to anyone stumbling across this of where new migrations should be implemented and any other changes they would need to make ie bumping the version number

**Review by keyallis — CHANGES_REQUESTED** [review]
This will need to be rebased for changes made in #4068

**Review by dashrews78 — COMMENTED** [review]
We may need to think about this one some more.  We have a situation where #4068 used a change in the patch release which hoses up migrations.  I have some ideas on how to work around it, but I think we need to know the Postgres version explicitly to do that.  Without storing it in the DB, I think we may be better served keeping PostgresDBVersionPlus, at least for now.  But I'm open to ideas.

File: migrator/migrations/registry.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ var (
 
 // MustRegisterMigration registers a Migration, panic-ing if there's an error.
 func MustRegisterMigration(m types.Migration) {
-	if !env.PostgresDatastoreEnabled.BooleanSetting() && m.StartingSeqNum > migrations.CurrentDBVersionSeqNumWithoutPostgres() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() && m.StartingSeqNum > migrations.BasePostgresDBVersionSeqNum() {
```
Comment (rhybrillou) [inline migrator/migrations/registry.go:17]: Does this mean non-postgres migrations beyond `LastRocksDBVersionSeqNum` will be silently ignored ?
If having extra non-postgres migrations is an error, wouldn't it be better to panic here, as suggested in the function documentation comment ?

File: pkg/migrations/internal/seq_num.go [hunk 2]
```diff
@@ -4,7 +4,8 @@ var (
 	// CurrentDBVersionSeqNum is the current DB version number.
 	// This must be incremented every time we write a migration.
 	// It is a shared constant between central and the migrator binary.
-	CurrentDBVersionSeqNum = 112
-	// PostgresDBVersionPlus is the current DB version number with Postgres DB data migration.
-	PostgresDBVersionPlus = 56
+	CurrentDBVersionSeqNum = 168
```
Comment (rhybrillou) [inline pkg/migrations/internal/seq_num.go:7]: Thank you for this semantic change. I think it makes things easier not to have to compute what the `PostgresDBVersionPlus` should be.

File: pkg/migrations/seq_num.go [hunk 3]
```diff
@@ -10,14 +10,14 @@ import (
...
 }
 
-// CurrentDBVersionSeqNumWithoutPostgres is the base of current DB version number
-// without Postgres migrations. This function should only be used in Postgres
+// BasePostgresDBVersionSeqNum is the base of DB version number
+// for Postgres migrations. This function should only be used in Postgres
 // migrations.
-func CurrentDBVersionSeqNumWithoutPostgres() int {
-	return internal.CurrentDBVersionSeqNum - 1
+func BasePostgresDBVersionSeqNum() int {
```
Comment (rhybrillou) [inline pkg/migrations/seq_num.go:21]: What is the lifetime of this function ? Should it be restrained to the initial Postgres DB load, or should it be used in newer migrations too, or could it be dropped so `CurrentDBVersionSeqNum` can be used directly in all n_ migrations ?

**Review by rhybrillou — COMMENTED** [review]
Thank you, I think this goes in the direction of making migration-writing easier.

File: pkg/migrations/seq_num.go (hunk 3 above)
Comment (c-du) [reply pkg/migrations/seq_num.go:21]: This should not be used in new Postgres to Postgres migrations. In your PR, I can make change accordingly after your PR merged.

**Review by keyallis — APPROVED** [review]
Unblocking

**Pull request merged**